### ISSUE TITLE
Timeline accessibility

### ISSUE THREAD

**Issue by qroll** [LifeSG/react-design-system#822]
**Component**
`Timeline`

**Describe the feature request**
- Align Timeline with the accessibility behaviour defined in the Figma specs
- Set the main title to render as a `h2` by default, and the item titles to `h3`
- Expose a `headingLevel` prop to customise the heading hierarchy. You may reference Accordion for the implementation.

**Acceptance criteria**
- Roles should be correctly read out by the AT
- Timeline item state should be read out by the AT

**References**
[Figma](https://www.figma.com/design/wDAxbRmlq4xOn4UJ8zPuEF/%F0%9F%A7%91%F0%9F%8F%BB%E2%80%8D%F0%9F%A6%AF-Flagship-A11y-Annotations?node-id=1617-7309&p=f&m=dev)
[Timeline Storybook](https://designsystem.life.gov.sg/react/index.html?path=/docs/feedback-indicators-timeline--docs)
[Accordion Storybook](https://designsystem.life.gov.sg/react/index.html?path=/docs/content-accordion--docs)
[Accordion source](https://github.com/LifeSG/react-design-system/tree/master/src/accordion)
